airbyte-webapp/src/components/base/Button/types.tsx

```diff
@@ -1,10 +1,22 @@
+import React from "react";
+
+export enum ButtonType {
+  Primary = "primary",
+  Secondary = "secondary",
+  Danger = "danger",
+  LightGrey = "lightGrey",
+}
+
 export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  customStyles?: string;
+  buttonType?: ButtonType;
+  clickable?: boolean;
   full?: boolean;
-  danger?: boolean;
-  secondary?: boolean;
+  icon?: React.ReactElement;
+  iconPosition?: "left" | "right";
   isLoading?: boolean;
-  iconOnly?: boolean;
+  label?: string | React.ReactNode;
+  size?: "xs" | "s" | "l";
```

Comment:
sizes abbreviations at least 2 letters, these should be `sm` and  `lg`. I would recommend also adding the `md` option and making that the default.